src/shogun/clustering/KMeans.cpp

```diff
@@ -80,32 +80,30 @@ void CKMeans::set_random_centers(SGVector<float64_t> weights_set, SGVector<int32
 	CDenseFeatures<float64_t>* lhs=
 		CDenseFeatures<float64_t>::obtain_from_generic(distance->get_lhs());
 
-	for (int32_t i=0; i<XSize; i++)
+	for (int32_t i=0; i<k; i++)
 	{
-		const int32_t Cl=CMath::random(0, k-1);
-		weights_set[Cl]+=1.0;
-		ClList[i]=Cl;
+		do
+		{
+			const int32_t Cl=CMath::random(0, XSize-1);
 
-		int32_t vlen=0;
+		} while(ClList[Cl]=Cl);
```

Comment:
1. Could you do this without a do-while?
2. Something is weird with the condition. You create random numbers until "hitting" something???
3. Please when changing things, make sure they are formatted according to our guidelines (even if the KMeans class itself is not)